### PR TITLE
BLOCK-1775 purchase window

### PR DESCRIPTION
**Pull request by justefg**

### CODE REVIEW

File: docs/contracts/NFT Contract/NFT Actions/setprchsreq.a.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ The factory manager can specify purchase options for users. Note that currently
 
     -   First, the cost in USD is (factory RAM payment size) \* (RAM price), where
 
-        -   NFT RAM payment size: **1656 bytes**
+        -   NFT RAM payment size: **1679 bytes**
```
Comment (igor-sikachyna) [inline docs/contracts/NFT Contract/NFT Actions/setprchsreq.a.md:25]: Also need to update the calculation below:

`(1661B/1024B \* 0.15USD/KB) / (conversion rate) = `0.24331055` **USD**/(conversion rate)`

File: docs/contracts/NFT Contract/NFT Actions/purchase.a.md [hunk 2]
```diff
@@ -74,6 +74,8 @@ Internally we are constructing a vector of which `token_ids` to be transferred.
 | receiver         | eosio::asset                        | string          | Account that will receive the Uniq from this purchase                                                                                                                               |
 | promoter_id      | std::optional\<eosio::name>         | string / null   | Optional promoter of the purchase transaction. If no promoter is provided then the default promoter specified in `saleshrlmcfg` (scope `0`) will be used if present                 |
 | user_uniqs       | std::optional\<provided_user_uniqs> | Array / null    | List of uniqs the buyer is willing to provide for this purchase option to either be taken from him or to just verify their presence. Refer to `provided_user_uniqs` breakdown below |
+| start            | std::optional\<time_point_sec>      | string / null          | Start time of purchase window (optional)
+| end            | std::optional\<time_point_sec>      | string / null          | End time of purchase window (optional)
```
Comment (igor-sikachyna) [inline docs/contracts/NFT Contract/NFT Actions/purchase.a.md:78]: The names used for the purchase period fields are not very clear. Better use purchase_window_start and purchase_window_end instead to clarify the purpose.

Also the formatting on the table is a bit broken. Ideally you should use some extension like this: https://marketplace.visualstudio.com/items?itemName=TakumiI.markdowntable

File: docs/contracts/NFT Contract/NFT Actions/setprchsreq.a.md [hunk 3]
```diff
@@ -95,6 +97,8 @@ cleos push action eosio.nft.ft setprchsreq.a '[
     },
     "sale_shares": [],
     "maximum_uos_payment": "2 UOS",
+    start: "2023-09-18T13:21:10.724",
+    end: "2023-11-18T13:21:10.724",
```
Comment (igor-sikachyna) [inline docs/contracts/NFT Contract/NFT Actions/setprchsreq.a.md:101]: Must put the `start` and `end` into quotes.

While at it, please update the `maximum_uos_payment` and `price` to use 8 digit prevision: "2.00000000 UOS"

File: docs/contracts/NFT Contract/NFT Actions/purchase.a.md [hunk 4]
```diff
@@ -128,6 +132,8 @@ await api.transact({
               strategy: 2
```
Comment (igor-sikachyna) [inline docs/contracts/NFT Contract/NFT Actions/purchase.a.md:132]: Same points as in `setprchsreq.a` page about UOS precision

File: docs/contracts/NFT Contract/NFT Actions/purchase.a.md (hunk 2 above)
Comment (justefg) [reply docs/contracts/NFT Contract/NFT Actions/purchase.a.md:78]: not sure why you say formatting is broken. looks good to me

https://github.com/ultraio/docs-blockchain/blob/feature/BLOCK-1775-purchase-window/docs/contracts/NFT%20Contract/NFT%20Actions/purchase.a.md

Comment (igor-sikachyna) [reply docs/contracts/NFT Contract/NFT Actions/purchase.a.md]: I mean not in the rendered doc, but the raw markdown view. The extension I linked adjusts the spacing in the markdown so it also looks like a normal table when looking at it even in the raw view

**Review by igor-sikachyna — APPROVED** [review]

**Pull request merged**